packages/types/src/type/Event.ts

```diff
@@ -86,18 +85,18 @@ export class EventIndex extends U8aFixed {
 export default class Event extends Struct {
   // Currently we _only_ decode from Uint8Array, since we expect it to
   // be used via EventRecord
-  constructor (_value?: Uint8Array) {
-    const { DataType, value } = Event.decodeEvent(_value);
+  constructor (_value: Uint8Array = new Uint8Array(), typeRegistry: TypeRegistry) {
```

Comment:
Yes, exactly the idea Jaco and I had some time ago: dependency injection via constructor. Teh tricky part is to figure out how to pass in `typeRegistry` to this constructor.

If i'm not mistaken, we instantiate this class here: https://github.com/polkadot-js/api/blob/master/packages/rpc-core/src/index.ts#L182. Which means that we somehow need to pass the typeRegistry to rpc-core too. I believe it's ultimately the correct way to go. We basically need to have `typeRegistry` be passed around in a lot of function arguments.